qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -196,15 +196,6 @@ def verify_schedule(self, dir_path, schedules, retentions=[]):
             self.assertTrue(retention in json_res['retention'])
 
 class TestSnapSchedules(TestSnapSchedulesHelper):
-    def remove_snapshots(self, dir_path):
```

Comment:
@mchangir I think with this change, remove_snapshots() from TestSnapSchedulesHelper class in getting invoked and it seems that there is a bug in that implementation as I mentioned in slack thread: https://ceph-storage.slack.com/archives/C04LVQMHM9B/p1746538360356159

```
    def remove_snapshots(self, dir_path, sdn):
        snap_path = f'{dir_path}/{sdn}'

        snapshots = self.mount_a.ls(path=snap_path)
        for snapshot in snapshots:
            if snapshot.startswith("_scheduled"):
                continue
            snapshot_path = os.path.join(snap_path, snapshot)
            log.debug(f'removing snapshot: {snapshot_path}')
            self.mount_a.run_shell(['sudo', 'rmdir', snapshot_path])
```

The check is for "_scheduled" and should be "scheduled-" IMO.

Reply:
... and the test case is racy as I mentioned in slack. See failure: https://pulpito.ceph.com/vshankar-2025-04-08_07:32:38-fs-wip-vshankar-testing-20250407.173548-debug-testing-default-smithi/8230961

Reply:
> If the snapshot name starts with `_scheduled`, we skip it with `continue`, else we delete any other snapshot.

So user snapshot names cannot start with an "_", which means this is skipping MDS internal snap names (long names) which show up under sub-directory snapdir if parent directories have been snapshotted. Is that correct?

Reply:
So, just to be clear

- the check for `_sheduled` is for internal snapshot names (snapshots could be taken on the parent directory/group and those will show up in sub-directories)
- the snap schedule is deactivated since that was racing with snapshot listing and removal of the directory.

Right? @mchangir

Reply:
@vshankar both inferences are correct